modules/gapi/test/gapi_statefulkernel_test_utils.hpp

```diff
@@ -0,0 +1,46 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2020 Intel Corporation
+
+#include "test_precomp.hpp"
+
+namespace opencv_test
+{
+template<typename T> struct get_first_param;
+template<typename R, typename P1> struct get_first_param<R(P1)> {
+  typedef P1 type;
+};
+
+#define DEFINE_INITIALIZER(Name, StateType, ...)                   \
+struct Name                                                        \
+{                                                                  \
+    static typename get_first_param<void(StateType)>::type value() \
+    {                                                              \
+       return __VA_ARGS__;                                         \
+    }                                                              \
+}                                                                  \
```

Comment:
????

Reply:
It is needed for easier creation of initializers for different state types.
Particularly get_first_param logic is responsible for possibility of such construction with commas:
DEFINE_INITIALIZER(StdArrayValue, (std::array<int, 3>), { 1, 2, 3 });

Reply:
Sounds over-designish, but lets have a look how it looks like in tests.